internal/middleware/authentication.go

```diff
@@ -53,6 +53,44 @@ func (m AuthenticationMiddleware) handleError(c *gin.Context, e error) {
 	_ = c.AbortWithError(http.StatusUnauthorized, e)
 }
 
+func (m AuthenticationMiddleware) QueryStringAuthentication(c *gin.Context) {
+	token, ok := c.GetQuery("token")
+	if !ok {
+		_ = c.Error(errdef.NewUnauthorized("token not valid"))
+		c.Abort()
+		return
+	}
+
+	user, err := parseToken(token, m.publicKey)
+	if err != nil {
+		_ = c.Error(errdef.NewUnauthorized("token not valid: %s", err.Error()))
+		c.Abort()
+		return
+	}
+
+	// Extra precaution to ensure that no errors has occurred, and it's safe to call c.Next()
+	if len(c.Errors.Errors()) > 0 {
+		c.Abort()
+		return
+	} else {
```

Comment:
Long time ago we had a bug granting people access in case of an error. This is my way of making it explicit that we're doing one OR the other. I'm obviously open for suggestions about to improve the code but perhaps this is more suited for an actual conversation.

Reply:
I was only saying that you do not need an else block. If you have a return in the `if` like you have you do not need an `else`.